### PR TITLE
refactor(pointcloud_preprocessor): prefix package and namespace with autoware

### PR DESCRIPTION
**Pull request by amadeuszsz**
## Description

Most of packages applies `pointcloud_preprocessor` plugin, therefore splitting PR for namespace and package prefix won't help with omitting breaking change. 

## Related links

**Parent Issue:**

- https://github.com/autowarefoundation/autoware/issues/4569

**Related PRs:**

- https://github.com/autowarefoundation/awsim_labs_sensor_kit_launch/pull/3
- https://github.com/autowarefoundation/sample_sensor_kit_launch/pull/95
- https://github.com/tier4/aip_launcher/pull/257
- https://github.com/RobotecAI/awsim_sensor_kit_launch/pull/15

## How was this PR tested?

Planning simulator & logging simulator

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: sensing/autoware_pointcloud_preprocessor/launch/distortion_corrector_node.launch.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   <!-- Parameter -->
   <arg name="param_file" default="$(find-pkg-share pointcloud_preprocessor)/config/distortion_corrector_node.param.yaml"/>
```
Comment (knzo25) [inline sensing/autoware_pointcloud_preprocessor/launch/distortion_corrector_node.launch.xml:8]: seems there is a missing prefix here

File: sensing/autoware_pointcloud_preprocessor/launch/blockage_diag.launch.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <arg name="vertical_bins" default="40"/>
   <arg name="is_channel_order_top2down" default="true"/>
   <arg name="blockage_diagnostics_param_file" default="$(find-pkg-share pointcloud_preprocessor)/config/blockage_diagnostics_param_file.yaml"/>
```
Comment (knzo25) [inline sensing/autoware_pointcloud_preprocessor/launch/blockage_diag.launch.xml:8]: missing prefix here?

File: sensing/autoware_pointcloud_preprocessor/launch/polygon_remover.launch.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ def generate_launch_description():
 
     my_component = ComposableNode(
         package=pkg,
```
Comment (knzo25) [inline sensing/autoware_pointcloud_preprocessor/launch/polygon_remover.launch.py:36]: the variable pkg does not have the prefix it seems

File: sensing/autoware_pointcloud_preprocessor/launch/preprocessor.launch.py [hunk 4]
```diff
@@ -36,7 +36,7 @@ def launch_setup(context, *args, **kwargs):
     if not is_separate_concatenate_node_and_time_sync_node:
         sync_and_concat_component = ComposableNode(
             package=pkg,
-            plugin="pointcloud_preprocessor::PointCloudConcatenateDataSynchronizerComponent",
```
Comment (knzo25) [inline sensing/autoware_pointcloud_preprocessor/launch/preprocessor.launch.py:39]: the variable pkg does not have the prefix it seems

**Comment by knzo25** [comment]
@amadeuszsz 
Could not leave a message, but `src/autoware/universe/sensing/autoware_pointcloud_preprocessor/launch/preprocessor.launch.xml` seems to be missing a prefix

File: sensing/autoware_pointcloud_preprocessor/launch/distortion_corrector_node.launch.xml (hunk 1 above)
Comment (amadeuszsz) [reply sensing/autoware_pointcloud_preprocessor/launch/distortion_corrector_node.launch.xml]: https://github.com/autowarefoundation/autoware.universe/pull/7983/commits/bf380f9d4f4d5284de7561e759f32b45684dd7b5

File: sensing/autoware_pointcloud_preprocessor/launch/blockage_diag.launch.xml (hunk 2 above)
Comment (amadeuszsz) [reply sensing/autoware_pointcloud_preprocessor/launch/blockage_diag.launch.xml]: https://github.com/autowarefoundation/autoware.universe/pull/7983/commits/3eac205f9b4aa6bda9edd41ffecdbd6d956d2a81

File: sensing/autoware_pointcloud_preprocessor/launch/polygon_remover.launch.py (hunk 3 above)
Comment (amadeuszsz) [reply sensing/autoware_pointcloud_preprocessor/launch/polygon_remover.launch.py:36]: https://github.com/autowarefoundation/autoware.universe/pull/7983/commits/77d9a06386901af0fe254d792fc1af3f87fdec27

File: sensing/autoware_pointcloud_preprocessor/launch/preprocessor.launch.py (hunk 4 above)
Comment (amadeuszsz) [reply sensing/autoware_pointcloud_preprocessor/launch/preprocessor.launch.py:39]: https://github.com/autowarefoundation/autoware.universe/pull/7983/commits/e19937d087db84bcb6fa0cf9a566eec8031b84ac

**Comment by amadeuszsz** [comment]
> @amadeuszsz Could not leave a message, but `src/autoware/universe/sensing/autoware_pointcloud_preprocessor/launch/preprocessor.launch.xml` seems to be missing a prefix

https://github.com/autowarefoundation/autoware.universe/pull/7983/commits/34c2e488ac6efe8f34ed0c1c4f6e5f7e2a70262e

**Review by knzo25 — APPROVED** [review]
LGTM :tada:

**Comment by amadeuszsz** [comment]
Friendly ping for package maintainers:
* @tkimura4 - `control/autoware_autonomous_emergency_braking`, `launch/tier4_planning_launch`
* @SakodaShintaro  - `launch/tier4_localization_launch`
* @yukkysaito  - `launch/tier4_perception_launch`, `perception/compare_map_segmentation`, `perception/euclidean_cluster`, `perception/ground_segmentation`, `perception/occupancy_grid_map_outlier_filter`, `perception/probabilistic_occupancy_grid_map`

**Review by tkimura4 — APPROVED** [review]
LGTM

**Comment by SakodaShintaro** [comment]
@amadeuszsz 
Thank you very much.  LGTM for `launch/tier4_localization_launch`

This is outside the scope of my review, but I have one comment.
According to [the documentation](https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/ros-nodes/directory-structure/), the recommended directory structure is `autoware_{name}/include/autoware/{name}`, but in this pull request, it seems to be `autoware_pointcloud_preprocessor/include/autoware_pointcloud_preprocessor`. Is this intentional?

**Review by SakodaShintaro — APPROVED** [review]

**Comment by amadeuszsz** [comment]
> @amadeuszsz Thank you very much. LGTM for `launch/tier4_localization_launch`
> 
> This is outside the scope of my review, but I have one comment. According to [the documentation](https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/ros-nodes/directory-structure/), the recommended directory structure is `autoware_{name}/include/autoware/{name}`, but in this pull request, it seems to be `autoware_pointcloud_preprocessor/include/autoware_pointcloud_preprocessor`. Is this intentional?

@SakodaShintaro thanks four your review and comment.
I've seen other PRs which were accepted under same directory structure convention (e.g. https://github.com/autowarefoundation/autoware.universe/pull/7892), thus I assumed there is no difference.
@technolojin could you please share your opinion here? If it's needed to follow `autoware_{name}/include/autoware/{name}` convention, I will do necessary changes.

**Comment by amadeuszsz** [comment]
@knzo25 @SakodaShintaro 
I did changes with reference to directory structure. Renaming will make commits non-trackable, therefore I did two commits:
* [soft](https://github.com/autowarefoundation/autoware.universe/pull/7983/commits/a07a2a96d1f61a0b80092ee2c2eea3d39b0ae312) - code changes
* [hard](https://github.com/autowarefoundation/autoware.universe/pull/7983/commits/f5879888c1b3073ad6d54825ecc1a4b21079812b) - only changed directory structure from `include/autoware_pointcloud_preprocessor` to `include/autoware/pointcloud_preprocessor`

**Comment by amadeuszsz** [comment]
@YoshiRi @badai-nguyen may I ask you for review remaining packages?

* @YoshiRi  - `launch/tier4_perception_launch`, `perception/ground_segmentation`, `perception/occupancy_grid_map_outlier_filter`, `perception/probabilistic_occupancy_grid_map`
* @badai-nguyen  -  `perception/compare_map_segmentation`, `perception/euclidean_cluster`, `perception/ground_segmentation`

**Comment by technolojin** [comment]
@amadeuszsz 

> @technolojin could you please share your opinion here? If it's needed to follow autoware_{name}/include/autoware/{name} convention, I will do necessary changes.

I am sorry for late response. It looks already done right.

**Review by technolojin — APPROVED** [review]
In terms of `tier4_perception_launch`, LGTM

**Review by YoshiRi — APPROVED** [review]
LGTM

**Pull request merged**